staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -214,7 +223,18 @@ func (a *auditResponseWriter) Write(bs []byte) (int, error) {
 	// the Go library calls WriteHeader internally if no code was written yet. But this will go unnoticed for us
 	a.processCode(http.StatusOK)
 	a.setHttpHeader()
-	return a.ResponseWriter.Write(bs)
+
+	// it is worth noting that Write method stores data in memory
+	// the default buffer size is 4096 K so most of the time this method won't return any errors
+	//
+	// one way of improving it would be changing the default buffer size - not supported by the std lib at the moment.
```

Comment:
I've opened https://github.com/openshift/kubernetes/pull/529 to experiment with the buffer size.